### DISCUSSION TITLE
Allow the user to choose which graph library to use

### DISCUSSION THREAD

**Discussion by Stefano-Buzzoni** [pyroll-project/pyroll-core#230] [Q&A] [answered]
Hi folks, I was wondering if there was a way to choose which library PyRoll uses to plot graphs. Implicitly I know that pandas which is integrated into PyRoll uses matplotlib by default, but as soon as I downloaded plotly for other needs, it automatically decided to use it for its plots too. In pandas I have the possibility to use the following string (`pd.options.plotting.backend = 'matplotlib`') to choose the library, it is also possible to do this by acting on the Pandas used by PyRoll?

**Comment by sag-kalibreur** [comment, marked as answer]
Hello Stefano,

Plotly actually has the unpleasant property of establishing itself as the primary plot function. However, you can change the preferred plotting application in **PyRoll** with this command:

`pr.PLOTTING_BACKEND = 'matplotlib'`

So you can use either plotly or matplotlib

**Additionally you can use some styles for matplotlib:**

`plt.style.use('dark_background')`
`plt.style.use('bmh')`
`plt.style.use('grayscale')`
`plt.style.use('ggplot')`
`plt.style.use(“_classic_test_patch”)`
`plt.style.use(“default”)`
`plt.style.use(“fivethirtyeight”)`
`plt.style.use(“classic”)`

**Reply by Stefano-Buzzoni** [reply]
Hi Sag, 
thanks for your kind reply I've tried the string you wrote but it wasn't acknowledging the 'PLOTTING_BACKEND' property, so after a few minutes of trying I discovered that the correct string is actually 

`pr.core.PLOTTING_BACKEND = 'matplotlib'`

so 'core' is actually required to be accessed in order to subsequently access the 'PLOTTING_BACKEND'.

Grateful also for the the plus of the styles personalization

**Comment by sag-kalibreur** [comment]
Hello Stefano,

of course you always have to import the PyRoll Core first, then the string will also work, I thought that was clear. You always need the PyRoll core for all calculations.

**So, this is how it works completely:**

`import pyroll.core as pr`
`pr.PLOTTING_BACKEND = 'matplotlib'`

...then the string works

**Reply by Stefano-Buzzoni** [reply]
Hi Michael,
I'm currently importing pyroll-basic so that's why I had to make this correction to the string then, 

`import pyroll.basic as pr`
`pr.core.PLOTTING_BACKEND = 'matplotlib'`

I've tried to use pyroll-core for my calculations but it alwys gave me error somehow that's why I fell back to pyroll-basic

**Reply by RollingTechSupport** [reply]
Hi Stefano, 
`pyroll-core` will perform just very basic "calculations". It's more like an empty hull. 
So depending on the error you faced, it could just be possible, that a needed plugin was not present.

**Reply by Stefano-Buzzoni** [reply]
Yeah that's the very same conclusion I've come to after trying it lol. Anyway thanks for you reply and congrats for your project (PyRoll), I'm actually a software developer and I got to know your library thanks to a colleague of mine who had attended a conference in Germany in which you presented it, so now I'm helping them with their needs.

**Reply by sag-kalibreur** [reply]
Hi Stefano,
if in doubt, just post your code so we can understand why an error occurs...

**Reply by Stefano-Buzzoni** [reply]
yeah those are past events luckily which I resolved, but if anything I will for sure, thanks
